### PR TITLE
use remote.endpoint instead of sourcegraph.url when installed in Sourcegraph Editor

### PR DESCRIPTION
**Pull request by nicksnyder**
fixes https://github.com/sourcegraph/src/issues/1322

### CODE REVIEW

**Comment by emidoots** [comment]
LGTM, thanks.

Please follow the release instructions under https://github.com/sourcegraph/sourcegraph-vscode#development meticulously

**Review by emidoots — APPROVED** [review]

**Comment by nicksnyder** [comment]
@slimsag I pushed a commit to address @sqs feedback here: https://sourcegraph.slack.com/archives/C3ZQ996A2/p1516124980000491

**Review by felixfbecker — APPROVED** [review]

**Comment by emidoots** [comment]
Does this actually work @nicksnyder ? I don't see how it could.

The package.json means the default value for `sourcegraph.url` will always be used in the case of Sourcegraph Editor, unless the user explicitly overrides it with `""`, right?

**Comment by nicksnyder** [comment]
haven't tested yet, but I think you are right. If so, than the existing fallback logic to "URL" wouldn't work either.

**Comment by emidoots** [comment]
I think the proper solution is just to prefer `remote.endpoint` first, then respect `sourcegraph.url` per usual -- but leave package.json untouched as it is now.

**Comment by emidoots** [comment]
> If so, than the existing fallback logic to "URL" wouldn't work either.

I agree. But this has been broken for a while then, so changing it now only causes users more headache as they've most likely already fixed their configurations.

**Review by emidoots — CHANGES_REQUESTED** [review]
.

**Comment by nicksnyder** [comment]
My point was that we can remove support for "URL" since it never worked to begin with.

I pushed a commit to address the issue.

File: src/extension.ts [hunk 1]
```diff
@@ -57,11 +57,17 @@ async function gitBranch(repoDir: string) {
...
 function sourcegraphURL() {
-  // Support old "URL" key or new "url" key.
-  const url =
-    vscode.workspace.getConfiguration("sourcegraph").get<string>("url") ||
-    vscode.workspace.getConfiguration("sourcegraph").get<string>("URL");
+  let url = vscode.workspace.getConfiguration("sourcegraph").get<string>("url");
+  // Use remote.endpoint if it is configured and sourcegraph.url is not.
+  if (url === DEFAULT_URL) {
+    url = vscode.workspace.getConfiguration("remote").get<string>("endpoint") || DEFAULT_URL;
+  }
```
Comment (emidoots) [inline src/extension.ts]: Why not just do `remote.endpoint || sourcegraph.url` effectively? I don't like the `url === DEFAULT_URL` implementation here.

We don't have to advertise that we support `remote.endpoint` to VS Code users, and it saves us from having bugs where `DEFAULT_URL` here does not match what is in `package.json`.

Comment (emidoots) [reply src/extension.ts]: ie
```
  // Support "remote.endpoint" first (Sourcegraph Editor) and "sourcegraph.url" second (VS Code).
  const url =
    vscode.workspace.getConfiguration("remote").get<string>("endpoint") ||
    vscode.workspace.getConfiguration("sourcegraph").get<string>("url");
```

File: package.json [hunk 2]
```diff
@@ -52,7 +52,7 @@
           "type": ["string"],
           "default": "https://sourcegraph.com",
           "description":
-            "The Sourcegraph URL to use. Specify your Sourcegraph Server instance's root URL here, if applicable."
+            "The Sourcegraph URL to use. Specify your Sourcegraph Server instance's root URL here, if applicable. When this extension is installed in Sourcegraph Editor, 'remote.configration' is used instead."
```
Comment (emidoots) [inline package.json]: Typo `configration` here

File: package.json [hunk 3]
```diff
@@ -52,7 +52,7 @@
           "type": ["string"],
           "default": "https://sourcegraph.com",
           "description":
-            "The Sourcegraph URL to use. Specify your Sourcegraph Server instance's root URL here, if applicable."
+            "The Sourcegraph URL to use. Specify your Sourcegraph Server instance's root URL here, if applicable. When this extension is installed in Sourcegraph Editor, 'remote.configuration' is used instead."
```
Comment (emidoots) [inline package.json]: actually another typo: it says `remote.configuration` but the real value is `remote.endpoint`, no?

**Review by emidoots — APPROVED** [review]

**Comment by emidoots** [comment]
😄 lgtm -- be careful to follow the release instructions in the steps they say :)

**Comment by felixfbecker** [comment]
That process looks overly complicated
- changelog should be only in CHANGELOG.md or GH releases, not in two places
- `version` should just be read from package.json
- bumping, tagging, committing can just be done with `npm version`
- the publish could happen in CI so access is not needed
- even the whole release could happen in CI if semantic-release is used

**Comment by emidoots** [comment]
no arguments there, but it's what we have today unless someone wants to update it.

**Pull request merged**